src/machine/asm/mod.rs

```diff
@@ -76,110 +78,121 @@ impl Memory<u64> for Box<AsmCoreMachine> {
         // 3% so we are keeping this version.
         let mut current_addr = addr;
         while current_addr < addr + size {
-            let page = current_addr / RISCV_PAGESIZE;
-            if self.flags[page] & FLAG_FREEZED != 0 {
+            let page = current_addr / RISCV_PAGESIZE as u64;
+            if self.flags[page as usize] & FLAG_FREEZED != 0 {
                 return Err(Error::InvalidPermission);
             }
-            current_addr += RISCV_PAGESIZE;
+            current_addr += RISCV_PAGESIZE as u64;
         }
         fill_page_data(self, addr, size, source, offset_from_addr)?;
         current_addr = addr;
         while current_addr < addr + size {
-            let page = current_addr / RISCV_PAGESIZE;
-            self.flags[page] = flags;
-            current_addr += RISCV_PAGESIZE;
+            let page = current_addr / RISCV_PAGESIZE as u64;
+            self.flags[page as usize] = flags;
+            current_addr += RISCV_PAGESIZE as u64;
         }
         Ok(())
     }
 
-    fn fetch_flag(&mut self, page: usize) -> Result<u8, Error> {
-        if page < RISCV_PAGES {
-            Ok(self.flags[page])
+    fn fetch_flag(&mut self, page: u64) -> Result<u8, Error> {
+        if page < RISCV_PAGES as u64 {
+            Ok(self.flags[page as usize])
         } else {
             Err(Error::OutOfBound)
         }
     }
 
-    fn store_bytes(&mut self, addr: usize, value: &[u8]) -> Result<(), Error> {
+    fn store_bytes(&mut self, addr: u64, value: &[u8]) -> Result<(), Error> {
         // Out of bound check is already performed in check_permission
-        check_permission(self, addr, value.len(), FLAG_WRITABLE)?;
-        let slice = &mut self.memory[addr..addr + value.len()];
+        check_permission(self, addr, value.len() as u64, FLAG_WRITABLE)?;
+        let slice = &mut self.memory[addr as usize..addr as usize + value.len()];
         slice.copy_from_slice(value);
         Ok(())
     }
 
-    fn store_byte(&mut self, addr: usize, size: usize, value: u8) -> Result<(), Error> {
+    fn store_byte(&mut self, addr: u64, size: u64, value: u8) -> Result<(), Error> {
         check_permission(self, addr, size, FLAG_WRITABLE)?;
-        memset(&mut self.memory[addr..addr + size], value);
+        memset(
+            &mut self.memory[addr as usize..(addr + size) as usize],
+            value,
+        );
         Ok(())
     }
 
-    fn execute_load16(&mut self, addr: usize) -> Result<u16, Error> {
+    fn execute_load16(&mut self, addr: u64) -> Result<u16, Error> {
         check_permission(self, addr, 2, FLAG_EXECUTABLE)?;
         self.load16(&(addr as u64)).map(|v| v as u16)
     }
 
     fn load8(&mut self, addr: &u64) -> Result<u64, Error> {
-        let addr = *addr as usize;
-        if addr + 1 > self.memory.len() {
+        let addr = *addr as u64;
```

Comment:
Many redudant `as u64` in this file.

Reply:
Ooops I was relying too much on clippy, already resolved.